netlogo-core/src/main/nvm/AnonymousProcedure.scala

```diff
@@ -25,10 +26,11 @@ sealed trait AnonymousProcedure {
   val locals: Array[AnyRef]
   def checkAgentClass(context: Context, agentClassString: String): Unit = {
     val kind = context.agent.kind
-    if (!(((kind == AgentKind.Observer) && agentClassString.contains('O')) ||
-          ((kind == AgentKind.Turtle) && agentClassString.contains('T')) ||
-          ((kind == AgentKind.Patch) && agentClassString.contains('P')) ||
-          ((kind == AgentKind.Link) && agentClassString.contains('L')))) {
+
+    if (!(((kind == AgentKind.Observer) && agentClassString(0) == 'O') ||
```

Comment:
Before:
```
observer> repeat 3 [ reset-timer ask one-of patches [ repeat 100000000 [ run [->] ] ] print timer ]
12.276
12.311
12.544
```

After:
```
observer> repeat 3 [ reset-timer ask one-of patches [ repeat 100000000 [ run [->] ] ] print timer ]
11.809
11.748
11.657
```

This was not particularly significant, but meh, I'll take it. Thanks for the suggestion!